layouts/shortcodes/event-group.html

```diff
@@ -14,20 +14,24 @@
 
 {{ $base_path := "event-schemas/schema" }}
 
-{{ $events := index .Site.Data "event-schemas" "schema" }}
 {{ $group_name := .Params.group_name }}
 
-{{ range $event_name, $event_data := $events }}
-
+{{/* Filter events and prepare them for sorting */}}
+{{ $events := slice }}
+{{ range $event_name, $event_data := index .Site.Data "event-schemas" "schema" }}
     {{ $prefix := substr $event_name 0 (len $group_name) }}
     {{ if eq $prefix $group_name }}
+        {{ $events = $events | append (dict "event_name" $event_name "event_data" $event_data) }}
+    {{ end }}
+{{ end }}
 
-        {{ $path := delimit (slice $base_path $event_name) "/" }}
-        {{ $event_data = partial "json-schema/resolve-refs" (dict "schema" $event_data "path" $path) }}
-        {{ $event_data := partial "json-schema/resolve-allof" $event_data }}
+{{/* Render the events sorted by x-weight or otherwise their event name */}}
+{{ range sort $events "event_data.x-weight" "event_name" }}
```

Comment:
Does this work? It looks like the only thing you can put as the third argument to `sort` is `"asc"` or `"desc"`: https://gohugo.io/functions/collections/sort/

Reply:
My bad and thanks for catching. I thought this worked because the output matched what I expected but it turns out this was just coincidence. The `sort` function silently [defaults](https://github.com/gohugoio/hugo/blob/4985be1a4af34dd835e7882605e429cd5ffb1912/tpl/collections/sort.go#L69) to `"asc"` if the second argument is anything other than `"desc"`.

I believe there isn't a strong use case for ties anyway so I simply made it sort by `x-weight` only and rebased the weights of call events to not cause ties.